### PR TITLE
Part4

### PR DESCRIPTION
**Pull request by hagansmith**

### CODE REVIEW

File: javascripts/controllers/favorites.js [hunk 1]
```diff
@@ -1,6 +1,39 @@
...
+    ContactService.getFavoriteContacts($rootScope.uid).then((results) => {
+    $scope.contacts = Object.keys(results).length === 0? null : results;
+    }).catch((error) => {
+      console.log("error in getContact", error);
+    });
+  };
+
+  getFavoriteContacts();
+
+  // $scope.deleteContact = (contact) => {
```
Comment (morecallan) [inline javascripts/controllers/favorites.js]: AHHHH COMMENTED OUT CODE! GET THIS OUT OF HERE!

File: partials/contacts/favorites.html [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+      </button>
+    <div class="collapse" id="{{key}}">
+      <h4>Cell Phone: {{contact.cell_phone}}</h4> <h4>Work Phone: {{contact.work_phone}}</h4>
+      <h4>Email: {{contact.email}}</h4>
+      <h5>{{contact.home_address}}</h5>
+      <h5>{{contact.home_city}}, {{contact.home_state}} {{contact.home_zip}}</h5>
+
+      <a class="favorite" ng-show="!contact.favorite" ng-click="updateFavorite(contact)"><i class="fa fa-star-o fa-2x"></i></a>
+      <a class="favorite" ng-show="contact.favorite" ng-click="updateFavorite(contact)"><i class="fa fa-star fa-2x"></i></a>
+      <!-- <button class="btn btn-default delete" ng-click="deleteContact(key)">Delete</button> -->
```
Comment (morecallan) [inline partials/contacts/favorites.html]: nooooooooo

**Review by morecallan — COMMENTED** [review]
@hagansmith - mostly minor stuff and looking pretty good! There is one bug: if you go to the edit page and make no adjustments and hit save, you get an error. Fix dat and push up with no commented out code 😉

**Comment by hagansmith** [comment]
@morecallan updated.

**Review by morecallan — COMMENTED** [review]
👍

**Pull request merged**